bgs/bgs.go

```diff
@@ -477,6 +484,59 @@ func (s *BGS) createExternalUser(ctx context.Context, did string) (*models.Actor
 		return nil, fmt.Errorf("claimed handle did not match servers response (%s != %s)", res.Did, did)
 	}
 
+	actinfo, err := s.lockedCreateUser(ctx, handle, did, peering.ID)
+	if err != nil {
+		if err == ErrRevalidateUserHandle {
+			var ou User
+			if err := s.db.First(&ou, "handle = ?", handle).Error; err != nil {
+				return nil, fmt.Errorf("failed to find pre-existing user with given handle after user creation failure: %w", err)
+			}
+
+			if ou.Did == did {
+				// What? How?
+				return nil, fmt.Errorf("failed to create user, but they already existed (very weird): did=%s, handle=%s", did, handle)
+			} else {
+				// a different user with the same handle?
+				doc, err := s.didr.GetDocument(ctx, ou.Did)
+				if err != nil {
+					return nil, err
+				}
+
+				ohandle, err := handleFromDidDoc(doc)
+				if err != nil {
+					return nil, err
+				}
+
+				if ohandle != ou.Handle {
+					// Aha! A handle change we didn't get a notice about
+					if err := s.db.Model(User{}).Where("id = ?", ou.ID).Update("handle", ohandle).Error; err != nil {
+						return nil, err
+					}
+
+					// TODO: it is a bit weird that we keep the handle in multiple locations...
+					if err := s.db.Model(models.ActorInfo{}).Where("uid = ?", ou.ID).Update("handle", ohandle).Error; err != nil {
+						return nil, err
+					}
+
+					// Now that thats resolved, try creating the user again?
+					return s.lockedCreateUser(ctx, handle, did, peering.ID)
```

Comment:
Small thing, but I think we should log even the happy path when this happens (not just error paths). Eg, we should log that it seems like handle changed, and include the prior handle and DID.